### DISCUSSION TITLE
Timeline navigation probably has a problematic bug.

### DISCUSSION THREAD

**Discussion by mikosenigma** [friction2d/friction#84] [General]
Hi, I wanted to ask about a problem with playing animations. During playback, the animation must be played in its entirety and only then I have the option to play it again. I can stop the playback at any time, but when I try to go back to the beginning and when I press play, the animation does not play from frame 0 but from the point at which I stopped it previously. I don't know if I'm explaining this clearly enough. Is this planned or is it a problem with the Linux version? My system is Ubuntu 22.04 Hardware specs. Ryzen5 5600 x processor and rx 6600 xt graphics card. I checked it on the latest stable version 0.9.4. Much respect to the developers! Who can I approach with an offer to help promote the project?

**Comment by rodlie** [comment]
Hi,

Not a bug, just lack of documentation and/or usability.

![friction-timeline-widgets](https://github.com/friction2d/friction/assets/34516798/0d836ddf-eac5-4c21-b6fa-c26054952724)

1. Go to first frame (SHIFT+KEY_LEFT)
2. Play from first frame
3. Go to last frame (SHIFT+KEY_RIGHT)
4. Play from current frame (SPACE)
5. Stop and return to frame where play was triggered (ESC)
6. Loop

Note that shortcut SPACE does different things depending on the state of the player.

- If stopped play from current frame (4)
- If playing pause (4)
- If pause play from current frame (4)

Maybe SHIFT+SPACE should be used as a shortcut to 2? Or something else, open for suggestions.

> the animation must be played in its entirety and only then I have the option to play it again

The content must be cached (rendered) before playback, see progress bar.

![Screenshot from 2023-09-19 15-24-16](https://github.com/friction2d/friction/assets/34516798/0823d86f-5788-48bf-980c-84ae38131f82)
